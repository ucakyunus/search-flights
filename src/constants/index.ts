export enum Currency {
  TRY = 'TRY',
}

export enum FareCategories {
  BUSINESS = 'BUSINESS',
  ECONOMY = 'ECONOMY',
}

export enum BrandCode {
  ECO_FLY = 'ecoFly',
  EXTRA_FLY = 'extraFly',
  PRIME_FLY = 'primeFly',
}

export enum Status {
  AVAILABLE = 'AVAILABLE',
  ERROR = 'ERROR',
}

export enum ListSortTypes {
  PriceAscending = 'PriceAscending',
  PriceDescending = 'PriceDescending',
  DepartureTimeAscending = 'DepartureTimeAscending',
  DepartureTimeDescending = 'DepartureTimeDescending',
}

export enum ListSortBy {
  DepartureTime = 'DepartureTime',
  Price = 'Price',
}

export enum ResultTypes {
  Success = 'Success',
  Failure = 'Failure',
}

export enum CounterActions {
  Increment = 'INCREMENT',
  Decrement = 'DECREMENT',
}
