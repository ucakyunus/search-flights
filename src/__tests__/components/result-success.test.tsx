import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Success, { SuccessProps } from '@/components/result/success';
import { Currency } from '@/constants';

const mockProps: SuccessProps = {
  amount: 100,
  currency: Currency.TRY,
};

describe('ResultSuccess Component', () => {
  it('ResultSuccess matches snapshot', () => {
    const { asFragment } = render(<Success {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct amount and currency', () => {
    render(<Success {...mockProps} />);
    expect(screen.getByText(/TRY\s*100/i)).toBeInTheDocument();
  });

  it('displays custom amount and currency values', () => {
    render(<Success amount={200} currency='EUR' />);
    expect(
      screen.getByText((content) => content.includes('200') && content.includes('EUR')),
    ).toBeInTheDocument();
  });

  it('displays large amount and different currency formats correctly', () => {
    render(<Success amount={1234567.89} currency='JPY' />);
    expect(
      screen.getByText((content) => content.includes('1234567.89') && content.includes('JPY')),
    ).toBeInTheDocument();
  });
});
