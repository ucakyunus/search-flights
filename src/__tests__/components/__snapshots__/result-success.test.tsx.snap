// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultSuccess Component ResultSuccess matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-1ri905o"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-y0m7b1-MuiSvgIcon-root"
        data-testid="CheckCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 css-zp3nss-MuiTypography-root"
      >
        Kabin seciminiz tamamlandi.
      </p>
    </div>
    <div
      class="MuiBox-root css-7wisrc"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-12i2tiu-MuiTypography-root"
      >
        Toplam Tutar
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-3k8414-MuiTypography-root"
      >
        TRY 100 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
